packages/adapter/src/lib/adapter/adapter.js

```diff
@@ -6762,9 +6765,9 @@ class AdapterClass extends EventEmitter {
                 try {
                     const objs = await this.getForeignObjectsAsync(pattern, null, null, options);
                     const promises = [];
-                    if (!this.aliasPatterns.includes(pattern)) {
+                    if (!this.aliasPatterns.has(pattern)) {
                         // its a new pattern to store
-                        this.aliasPatterns.push(pattern);
+                        this.aliasPatterns.add(pattern);
                     }
```

Comment:
That's not necessary with a Set, just add the pattern. If it already exists, that won't do anything.

Reply:
yes true, however I really find it easier to read with the ceck, but if you think it is important performance wise I can remove the check in the two places it occurs and adjust the comment, because it is then not for sure a new pattern

Reply:
I don't think it matters performance wise. Set lookups, both for `has` and `add` are O(1)